universal-application-tool-0.0.1/test/BrowserTest.java

```diff
@@ -50,5 +50,25 @@ public void noCredLogin() {
     assertTrue(browser.pageSource().contains("GuestClient"));
     assertTrue(browser.pageSource().contains("{\"created_time\":"));
     assertTrue(browser.pageSource().contains(Roles.ROLE_APPLICANT.toString()));
+    browser.goTo(baseUrl + controllers.admin.routes.AdminProgramController.index().url());
+    assertTrue(browser.pageSource().contains("403"));
+  }
+
+  @Test
+  public void adminTestLogin() {
+    String baseUrl = "http://localhost:" + play.api.test.Helpers.testServerPort();
+    browser.goTo(baseUrl + routes.HomeController.loginForm(Optional.empty()).url());
+    browser.$("#admin").click();
+    // should be redirected to root.
+    assertEquals("", browser.url());
+    assertTrue(browser.pageSource().contains("Your new application is ready."));
+    browser.goTo(baseUrl + routes.HomeController.secureIndex().url());
+    assertTrue(browser.pageSource().contains("You are logged in."));
+    browser.goTo(baseUrl + routes.ProfileController.myProfile().url());
+    assertTrue(browser.pageSource().contains("FakeAdminClient"));
+    assertTrue(browser.pageSource().contains("{\"created_time\":"));
+    assertTrue(browser.pageSource().contains(Roles.ROLE_UAT_ADMIN.toString()));
+    browser.goTo(baseUrl + controllers.admin.routes.AdminProgramController.index().url());
+    assertTrue(browser.pageSource().contains("Programs"));
```

Comment:
Suggestion: To make the test more readable, add whitespace and consider adding comments here and there about what's going on / what's being tested, as it makes sense. It's not obvious to me (without separate titled tests) what we are semantically testing on each `browser.goTo`. (I assume separating these out is not performant and/or the ordered steps are crucial for the test).

```suggestion
    String baseUrl = "http://localhost:" + play.api.test.Helpers.testServerPort();
    
    browser.goTo(baseUrl + routes.HomeController.loginForm(Optional.empty()).url());
    browser.$("#admin").click();
    // should be redirected to root.
    assertEquals("", browser.url());
    assertTrue(browser.pageSource().contains("Your new application is ready."));
    
    browser.goTo(baseUrl + routes.HomeController.secureIndex().url());
    assertTrue(browser.pageSource().contains("You are logged in."));
    
    browser.goTo(baseUrl + routes.ProfileController.myProfile().url());
    assertTrue(browser.pageSource().contains("FakeAdminClient"));
    assertTrue(browser.pageSource().contains("{\"created_time\":"));
    assertTrue(browser.pageSource().contains(Roles.ROLE_UAT_ADMIN.toString()));
    
    browser.goTo(baseUrl + controllers.admin.routes.AdminProgramController.index().url());
    assertTrue(browser.pageSource().contains("Programs"));
```